src/Mgate/PubliBundle/Form/Type/RelatedDocumentType.php

```diff
@@ -22,39 +22,43 @@ public function buildForm(FormBuilderInterface $builder, array $options)
-                ));
+                'class' => 'Mgate\SuiviBundle\Entity\Etude',
+                'choice_label' => 'nom',
+                'required' => false,
+                'label' => 'Document lié à l\'étude',
+                'data' => $options['etude'],
+                'attr' => array('style' => 'min-width: 300px'),
+                'configs' => array('placeholder' => 'Sélectionnez une étude', 'allowClear' => true),
+            ));
         }
         if ($options['prospect']) {
             $builder->add('prospect', Select2EntityType::class, array(
-                    'class' => 'Mgate\PersonneBundle\Entity\Prospect',
-                    'choice_label' => 'nom',
-                    'required' => false,
-                    'label' => 'Document lié au prospect',
-                    'configs' => array('placeholder' => 'Sélectionnez un prospect', 'allowClear' => true),
-                ));
+                'class' => 'Mgate\PersonneBundle\Entity\Prospect',
+                'choice_label' => 'nom',
+                'required' => false,
+                'label' => 'Document lié au prospect',
+                'attr' => array('style' => 'min-width: 300px'),
+                'configs' => array('placeholder' => 'Sélectionnez un prospect', 'allowClear' => true),
+            ));
         }
         if ($options['formation']) {
             $builder->add('formation', Select2EntityType::class, array(
-                    'class' => 'Mgate\FormationBundle\Entity\Formation',
-                    'choice_label' => 'titre',
-                    'required' => false,
-                    'label' => 'Document lié à la formation',
-                    'configs' => array('placeholder' => 'Sélectionnez une formation', 'allowClear' => true),
-                ));
+                'class' => 'Mgate\FormationBundle\Entity\Formation',
+                'choice_label' => 'titre',
+                'required' => false,
+                'label' => 'Document lié à la formation',
+                'attr' => array('style' => 'min-width: 300px'),
+                'configs' => array('placeholder' => 'Sélectionnez une formation', 'allowClear' => true),
+            ));
         }
         if ($options['etudiant'] || $options['etude']) {
             $builder->add('membre', Select2EntityType::class, array(
-                    'label' => 'Document lié à l\'étudiant',
-                    'class' => 'Mgate\\PersonneBundle\\Entity\\Membre',
-                    'choice_label' => 'personne.prenomNom',
-                    'required' => false,
-                    'configs' => array('placeholder' => 'Sélectionnez un étudiant', 'allowClear' => true), ))
-        ;
+                'label' => 'Document lié à l\'étudiant (optionnel)',
+                'class' => 'Mgate\\PersonneBundle\\Entity\\Membre',
+                'choice_label' => 'personne.prenomNom',
+                'required' => false,
+                'attr' => array('style' => 'min-width: 300px'),
```

Comment:
I agree with that rules and try to apply it in most of the cases.  (`DocType/upload.html.twig` in that PR for instance)

In this situation, we have just one template `Document/upload.html.twig` which is used by several document upload forms. 
Thus template has no idea about the form structure and renders it programatically (just iterate on form fields). 
And apart from checking form name in the template (heavy and not very adaptative), I don't see any other solution to display fields with the appropriate styling.